CMake logic bug in MOM6_GEOSPlug

From @amdasilva
```CMake Error at @cmake/@ecbuild/cmake/ecbuild_log.cmake:168 (message):
  ERROR - Directory not found mom6_cmake (possibly sparse checkout)
Call Stack (most recent call first):
  @cmake/esma_support/esma_mepo_style.cmake:29 (ecbuild_error)
  src/Components/@GEOSgcm_GridComp/GEOSogcm_GridComp/GEOS_OceanGridComp/MOM6_GEOSPlug/mom6_cmake/CMakeLists.txt:463 (esma_mepo_style)
```
The following line should have used the optional argument FOUND in the macro call:
```cmake
esma_mepo_style(mom6 MOM6_rel_path REL_PATH ..)
```
Without that, one cannot safely handle the situation in which a "sparsable" directory is missing.

The fix is a bit nontrivial in this case, as we usually stub at the component level, but here we are trying to stub one level down.

The first fix is to change the line above to:
```cmake
esma_mepo_style(mom6 MOM6_rel_path REL_PATH .. FOUND mom6_found)
```
It should then be moved to near the top of the file and a conditional should be added:
```
if (mom6_found)
   <<remaining accumulation of SRC files>
else ()
   esma_create_stub_component MOM6_SRCS mom6)
endif ()
...
```
Getting the details right will require a bit of direct examination of the details here.

--- a/GEOSogcm_GridComp/GEOS_OceanGridComp/CMakeLists.txt
+++ b/GEOSogcm_GridComp/GEOS_OceanGridComp/CMakeLists.txt
@@ -1,10 +1,7 @@
 esma_set_this ()
 
-add_subdirectory(MOM6_GEOSPlug)
-add_subdirectory(MOM_GEOS5PlugMod)
-add_subdirectory(GEOSdatasea_GridComp)
-
 esma_add_library (${this}
   SRCS GEOS_OceanGridComp.F90
+  SUBCOMPONENTS MOM6_GEOSPlug MOM_GEOS5PlugMod GEOSdatasea_GridComp
   DEPENDENCIES GEOSdatasea_GridComp MAPL
   INCLUDES ${INC_ESMF})

